flang/lib/Lower/OpenMP.cpp

```diff
@@ -138,9 +138,10 @@ genOMP(Fortran::lower::AbstractConverter &converter,
   llvm::ArrayRef<mlir::Type> argTy;
   if (blockDirective.v == llvm::omp::OMPD_parallel) {
 
-    mlir::Value ifClauseOperand, numThreadsClauseOperand;
+    mlir::Value ifClauseOperand, numThreadsClauseOperand, allocator;
     SmallVector<Value, 4> privateClauseOperands, firstprivateClauseOperands,
-        sharedClauseOperands, copyinClauseOperands;
+        sharedClauseOperands, copyinClauseOperands, 
+        allocatorOperands, allocateOperands;
```

Comment:
`allocatorOperands` is `SmallVector`. Can there be multiple allocator within single clause ?

Reply:
In the Fortran syntax there can not be more than one allocator per allocate clause, however, that is different in the mlid implementation of the allocate clause. In order to keep track of which value have what allocator, I enforced that each value will have an allocator, event if one is not specified(in this case it will be the default allocator). So yes, there will be multiple allocators within a single clause and the allocateOperands SmallVector keeps track of them.